clinica/utils/filemanip.py

```diff
@@ -1,209 +1,476 @@
+
+    .. note::
+        If a list of file paths is provided, then this function
+        will be called on each item and a list of paths to the
+        resulting unzipped files will be returned.
+        The type hint is not reflecting this fact because of the
+        self-contained requirement from Nipype which is preventing
+        the usage of types outside of Python's builtins.
+
+    Parameters
+    ----------
+    in_file: str
+        Path to file to be zipped.
+
+    same_dir: bool, optional
+        If True, the unzipped file is written in the same directory as the
+        provided zip file. Otherwise, it is written in the current working
+        directory. Default=False.
+
+    Returns
+    -------
+    out_file: str
+        Path to the resulting unzipped file.
+
+    Notes
+    -----
+    If the provided file is not zipped (that is, its extension
+    is different from '.gz'), then nothing is done and the path
+    to the provided file is returned.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the provided file does not exist.
+    """
+    from clinica.utils.filemanip import _zip_unzip_nii  # noqa
+
+    return _zip_unzip_nii(in_file, unzip_nii, same_dir)
 
-    if (in_file is None) or isinstance(in_file, _Undefined):
-        return None
 
-    if not isinstance(in_file, str):  # type(in_file) is list:
-        return [unzip_nii(f) for f in in_file]
+def save_participants_sessions(
+    participant_ids: List[str],
+    session_ids: List[str],
+    out_folder: str,
+    out_file: Optional[str] = "participants.tsv",
+) -> None:
+    """Save the participants and sessions to TSV.
 
-    _, base, ext = split_filename(in_file)
+    Parameters
+    ----------
+    participant_ids: List[str]
+        List of participant IDs.
 
-    # Not compressed
-    if ext[-3:].lower() != ".gz":
-        return in_file
-    # Compressed
-    gunzip = Gunzip(in_file=in_file)
-    gunzip.run()
-    return gunzip.aggregate_outputs().out_file
+    session_ids: List[str]
+        List of session IDs.
 
+    out_folder: str
+        Folder where the resulting TSV file should be written.
 
-def save_participants_sessions(participant_ids, session_ids, out_folder, out_file=None):
-    """Save <participant_ids> <session_ids> in <out_folder>/<out_file> TSV file."""
+    out_file: str, optional
+        Name of the TSV file. Default="participants.tsv".
+
+    Raises
+    ------
+    ValueError
+        If provided `participant_ids` and `session_ids` do not have
+        the same length.
+    """
     import os
+    from pathlib import Path
 
-    import pandas
+    import pandas as pd
 
     from clinica.utils.stream import cprint
 
-    assert len(participant_ids) == len(session_ids)
+    if len(participant_ids) != len(session_ids):
+        raise ValueError(
+            "The number of participant IDs is not equal to the number of session IDs."
+        )
 
+    out_folder = Path(out_folder)
     os.makedirs(out_folder, exist_ok=True)
```

Comment:
Using [Path.mkdir](https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir)

```suggestion
    out_folder.mkdir(parents=True, exist_ok=True)
```